src/plugins/WebOfTrust/util/TickerDelayedBackgroundJob.java

```diff
@@ -0,0 +1,370 @@
+package plugins.WebOfTrust.util;
+
+import freenet.node.FastRunnable;
+import freenet.node.PrioRunnable;
+import freenet.support.Executor;
+import freenet.support.Ticker;
+import freenet.support.io.NativeThread;
+
+/**
+ * A {@link DelayedBackgroundJob} implementation that uses a {@link Ticker} for scheduling.
+ *
+ * @author bertm
+ * @see TickerDelayedBackgroundJobFactory
+ */
+public class TickerDelayedBackgroundJob implements DelayedBackgroundJob {
+    static enum JobState {
+        /** Waiting for a trigger, no running job thread or scheduled job. */
+        IDLE,
+        /** Waiting for the scheduled job to be executed, no running job thread. */
+        WAITING,
+        /** Running the job in the job thread. */
+        RUNNING,
+        /** Waiting for the running job thread to finish before we terminate. */
+        TERMINATING,
+        /** Terminated, no running job thread. */
+        TERMINATED
+    }
```

Comment:
FYI, no change required:

When generally thinking about the experience while reviewing the class, I found it difficult to always remember the values which were expected to be valid for the various member variables in each state.
You already have a lot of assert(), which help a lot with the purpose of showing which values the member variables ought to be in which state.
However, the assert() are usually only in places where the state is *changed*. This makes it difficult to understand since the valid range of member variables also changes, and the state-changing functions can be called from multiple states.

A good way of documenting the valid ranges of the member variables might be to add a member function "validate()" or whatever to the enum JobState, which has the sole purpose of containing assert() about what the member variables of the TickerDelayedBackgroundJob should be in the *current* state, not in state transitions.
Something like:
```java
public boolean validate() {
  switch(this) {
    case IDLE:
      assert(waitingTickerJob == 0)
    ...
  }
  return true; // Return boolean so we can contain the call to this function in an assert() for performance
}
```

Then you can call (TickerDelayedBackgroundJob.this.)state.validateState() in many places where it might make sense, and possible also in unit tests.